pkg/volume/awsebs/attacher.go

```diff
@@ -175,15 +179,20 @@ func (attacher *awsElasticBlockStoreAttacher) WaitForAttach(spec *volume.Spec, d
 	for {
 		select {
 		case <-ticker.C:
-			klog.V(5).Infof("Checking AWS Volume %q is attached.", volumeID)
-			devicePaths := getDiskByIDPaths(aws.KubernetesVolumeID(volumeSource.VolumeID), partition, devicePath)
-			path, err := verifyDevicePath(devicePaths)
+			klog.V(5).Infof("Checking AWS Volume %q is attached at devicePath %q.", volumeID, devicePath)
+			path := ""
+			if runtime.GOOS == "windows" {
+				path, err = attacher.getDiskNumber(volumeID)
```

Comment:
I think it's better put windows related code into attacher_windows.go, and write a common function like `attacher.getDevicePath(volumeID)` for both linux and windows, so you don't need to check current code is running on windows or linux.

Reply:
@leakingtapan thought the same
Personally I don't like doing it just for 1 short function, and the ugly runtime.GOOS check is hard to avoid in other places anyway, but since you both agree I'll refactor it like that 👍

Reply:
refactored